src/recursive.rs

```diff
@@ -128,24 +147,36 @@ impl<'a, 'b, I: Input<'a>, O, E: ParserExtra<'a, I>> Recursive<Indirect<'a, 'b,
             inner: RecursiveInner::Owned(RefC::new(Indirect {
                 inner: OnceCell::new(),
             })),
+            state: EmptyPhantom::new(),
         }
     }
 
     /// Defines the parser after declaring it, allowing it to be used for parsing.
     // INFO: Clone bound not actually needed, but good to be safe for future compat
     #[track_caller]
-    pub fn define<P: Parser<'a, I, O, E> + Clone + MaybeSync + 'a + 'b>(&mut self, parser: P) {
+    pub fn define<P: Parser<'a, I, O, E> + Clone + MaybeSync + 'a + 'b>(
+        parser: P,
+        declaration: Self,
+    ) -> Recursive<Indirect<'a, 'b, I, O, E>, Defined> {
```

Comment:
What's the reason for not having the `declaration` argument be `self`?

Reply:
Honestly nothing in particular. I'll swap it over to having it be a method!

Reply:
@zesterer Which one do you prefer? I'll do it whichever way you'd prefer. I personally like the symmetry between `Recursive::declare` and `Recursive::define` and the second way makes it more obvious that `Recursive::define` consumes the parser.

Reply:
Hmm, I think I prefer the first, personally. The extra parameter way off at the end of the definition feels very isolated and random if you're not already sure what it is. Also, putting it at the front makes clear what the target of the definition is.